plugins/inputs/tail/tail.go

```diff
@@ -162,7 +169,32 @@ func (t *Tail) Start(acc telegraf.Accumulator) error {
 	return err
 }
 
-func (t *Tail) tailNewFiles(fromBeginning bool) error {
+func (t *Tail) getSeekInfo(file, initialReadOffset string) (*tail.SeekInfo, error) {
+	switch initialReadOffset {
```

Comment:
No need to pass the offset config?
```suggestion
func (t *Tail) getSeekInfo(file string) (*tail.SeekInfo, error) {
	switch t.InitialReadOffset {
```

Reply:
Currently `tailNewFiles` was called by `Gather` with `fromBeginning` set to `true` and called by `Start` with `t.FromBeginning`.
https://github.com/influxdata/telegraf/blob/4905e9f5b9cb90c71944ad877dda79e8222a854e/plugins/inputs/tail/tail.go#L124-L155
https://github.com/influxdata/telegraf/blob/4905e9f5b9cb90c71944ad877dda79e8222a854e/plugins/inputs/tail/tail.go#L165-L197

I passed this config so that `Gather` can let `tailNewFiles` to read from beginning.